File: Java/Queue/Queue.java
import java.util.Arrays;

class QueueArray {
    int front = -1;
    int rear = -1;
    int size = 10;
    int queueArray[] = new int[size];

    void overFlow() {
        System.out.println("Queue Overflow!");
    }

    void underFlow() {
        System.out.println("Queue is empty!");
    }

    void add(int value) {
        if (rear == size - 1) {
            overFlow();
            return;
        }
        if (front == -1) {
            front = 0;
        }
        queueArray[++rear] = value;
        System.out.println("Element Enqueued: " + value);
    }

    void remove() {
        if (front == -1 || front > rear) {
            underFlow();
            return;
        }
        int removedElement = queueArray[front++];
        System.out.println("Element Dequeued: " + removedElement);
        if (front > rear) {
            front = -1;
            rear = -1;
        }
    }

    int poll() {
        if (front == -1 || front > rear) {
            underFlow();
            return -1;
        }
        int removedItem = queueArray[front++];
        if (front > rear) {
            front = -1;
            rear = -1;
        }
        return removedItem;
    }

    int peek() {
        if (front == -1 || front > rear) {
            underFlow();
            return -1;
        }
        return queueArray[front];
    }

    int element() {
        if (front == -1 || front > rear) {
            throw new IllegalStateException("Queue is empty!");
        }
        return queueArray[front];
    }

    void display() {
        if (front == -1) {
            underFlow();
            return;
        }
        System.out.println(Arrays.toString(queueArray));
    }
}

public class Queue {
    public static void main(String[] args) {
        QueueArray queue = new QueueArray();

        queue.add(1);
        queue.add(2);
        queue.add(3);

        queue.display();

        queue.remove();
        queue.display();

        System.out.println("Peek: " + queue.peek());
        System.out.println("Poll: " + queue.poll());

        queue.display();

        System.out.println("Element: " + queue.element());

        queue.remove();
        queue.remove();
        queue.remove();
    }
}

File: Python/Queue/Queue.py
class Queue:
    def __init__(self,size) -> None:
        self.front: int = -1
        self.rear: int = -1
        self.size: int = size
        self.queueArray: list[int] = [0] * size
    
    def add(self,value:int) -> None:
        if self.rear >= self.size -1:
            print("Overflow Queue.")
            return
        
        if self.front == -1:
            self.front = 0

        self.rear += 1
        self.queueArray[self.rear] = value
    
    def remove(self) -> None:
        if self.front == -1 or self.front > self.rear:
            print("Underflow Queue.")
        
        self.queueArray[self.front] = 0
        self.front += 1

        if self.front > self.rear:
            self.front = - 1
            self.rear = - 1

    def showQueueArray(self,):
        print(self.queueArray)





q = Queue(10)
q.add(10)
q.add(11)
q.add(12)
q.add(13)
q.showQueueArray()

        
    